### PR TITLE
feat: diff function added to paddle frontend

### LINKED ISSUE #23701: diff

### PR DESCRIPTION
**Pull request by mintesin**
Close #23701 

- [x] Did you add a function?yes
- [x] Did you add the tests?yes

### Socials:
LinkedIn:https://www.linkedin.com/in/mintewargaw

### CODE REVIEW

**Comment by mintesin** [comment]
Hello @bipinKrishnan can you please help me with the errors that appeared on the above test? I can't Understand what went wrong and fix the Error saying Error: Process completed with exit code 1. Please Help!!!!

**Comment by mintesin** [comment]
@bipinKrishnan  I have 6 errors among them four of them are about "requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))" which is about request module. its remote connection is failing and Two of them are errors on unabling for importing dill library(on code space this  is working normally but not on github) any Idea please???

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -551,6 +551,13 @@ def equal(self, y, name=None):
     @with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")
     def rad2deg(self, name=None):
         return paddle_frontend.Tensor(ivy.rad2deg(self._ivy_array))
+    
+    @with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+    def diff(self, y, name=None):
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:556]: There are more args that `diff` function supports. You could refer [here](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/diff_en.html#diff)

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -551,6 +551,13 @@ def equal(self, y, name=None):
     @with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")
     def rad2deg(self, name=None):
         return paddle_frontend.Tensor(ivy.rad2deg(self._ivy_array))
+    
+    @with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:555]: You could refer to the supported dtypes in the paddle docs mentioned [here](https://github.com/unifyai/ivy/pull/23702/files#r1335037678)

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 3]
```diff
@@ -551,6 +551,13 @@ def equal(self, y, name=None):
...
+    def diff(self, y, name=None):
+        self.ivy_array = paddle_frontend.Tensor(
+            ivy.diff(self._ivy_array, _to_ivy_array(y))
+        ).ivy_array
+        return self
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:560]: Should return a tensor here

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 4]
```diff
@@ -422,6 +422,41 @@ def test_paddle_tensor_abs(
...
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="diff",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes('valid'),
+        num_arrays=2,
+        shared_dtype=True,
+    ),
+)
+def test_paddle__diff(
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:436]: Remove the double underscore

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 5]
```diff
@@ -422,6 +422,41 @@ def test_paddle_tensor_abs(
...
+    dtype_x_,
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:437]: Should be `dtype_and_x`

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py (hunk 4 above)
Comment (mintesin) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: double underscore removed

**Comment by mintesin** [comment]
@bipinKrishnan Thank you for your response. I have updated the tensor.py accordingly. Would you take a look at it now, please??

**Comment by bipinKrishnan** [comment]
ivy-gardener

**Comment by bipinKrishnan** [comment]
Hey, the lint tests are failing currently. Could you please install pre-commit in your system so that the code is linted correctly and push again?

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 6]
```diff
@@ -551,6 +551,23 @@ def equal(self, y, name=None):
...
+         "float64", 
+         "bool", 
+         "int32", 
+         "int64"
+        )
+      }, 
+     "paddle" 
+    )
+    def diff(self,y, n=1,axis=-1,prepend=None,append=None, name=None):
+        self.ivy_array = paddle_frontend.Tensor(
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:567]: ivy.diff returns ivy array, so no need multiple conversion here

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 7]
```diff
@@ -551,6 +551,23 @@ def equal(self, y, name=None):
...
+            ivy.diff(self._ivy_array, _to_ivy_array(y))
+        ).ivy_array
+        return paddle_frontend.Tensor(self)
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:570]: This should return an array if I'm correct

Comment (mintesin) [reply ivy/functional/frontends/paddle/tensor/tensor.py:570]: The documentation from paddle says it is Tensor

**Comment by bipinKrishnan** [comment]
Hey @mintesin, I've left some suggestions on the changes you've made.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py [hunk 8]
```diff
@@ -50,7 +50,8 @@ def _elemwise_helper(draw):
 # ------------ #
 
 
-# ToDo: Fix this test after torch override of assert is implemented
+# ToDo: Fix this test after torch overide of assert is implemented
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:53]: You could undo these changes as its not related to your PR.

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 9]
```diff
@@ -538,6 +538,23 @@ def equal(self, y, name=None):
...
+                "float32",
+                "float64",
+                "bool",
+                "int32",
+                "int64",
+            )
+        },
+        "paddle",
+    )
+    def diff(self_, y, n=1, axis=-1, prepend=None, append=None, name=None):
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:554]: It should be `self` instead of `self_`

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 10]
```diff
@@ -538,6 +538,23 @@ def equal(self, y, name=None):
...
+        self_ = paddle_frontend.diff(self_, y)
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:555]: You could use some simple but intuitive variable names instead of `self_` as its a bit confusing with `self`. May be refer to other implementations as well.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
